applications/flash_attention_v2/collective/xe_flash_attn_prefill_mma.hpp

```diff
@@ -180,18 +189,87 @@ struct FlashPrefillMma<gemm::MainloopIntelXeXMX16<Stages>, ProblemShapeType_, El
+
 
     int thread_idx = static_cast<int>(ThreadIdxX());
+    if constexpr (rope_enabled) {
+      // calculate the base_ptr and offset for Q, K.
+      // also calculate the layout for Q, K.
+      // then apply RoPE on Q, K accordingly
+      auto [coord_q_x, coord_q_y, coord_q_z] = *gQ.data();
+      auto [coord_k_x, coord_k_y, coord_k_z] = *gK.data();
+      
+      auto [batch, num_heads_q, num_heads_kv, seq_len_qo, seq_len_kv, head_size_qk, head_size_vo] = problem_shape;
+      
+      int offset_q = seq_len_qo*head_size_qk*coord_q_z + head_size_qk*coord_q_x + coord_q_y; // row major
+      // int offset_k = seq_len_kv*head_size_qk*coord_k_z + head_size_qk*coord_k_y + coord_k_x; // col major
+      int offset_k = seq_len_kv*head_size_qk*coord_k_z + head_size_qk*coord_k_x + coord_k_y; // row major
+
+      auto q_traits = static_cast<traits_load_Q const&>(params.gmem_tiled_copy_q);
+      ElementQ* base_ptr_q = (ElementQ*)q_traits.base_ptr;
+
+      auto q_traits_cos = static_cast<traits_load_Q const&>(params.gmem_tiled_copy_q_cos);
+      ElementQ* base_ptr_q_cos = (ElementQ*)q_traits_cos.base_ptr;
+
+      auto q_traits_sin = static_cast<traits_load_Q const&>(params.gmem_tiled_copy_q_sin);
+      ElementQ* base_ptr_q_sin = (ElementQ*)q_traits_sin.base_ptr;
+
+      // auto layout_q = gQ.layout();
+      constexpr auto static_shape_q = make_shape(size<0>(gQ), size<1>(gQ));
+      // constexpr auto layout_q = LayoutQ::packed({size<0>(gQ), size<1>(gQ)});
+      constexpr auto layout_q = make_layout(static_shape_q, LayoutRight{});
+      
+      auto k_traits = static_cast<traits_load_K const&>(params.gmem_tiled_copy_k);
+      ElementK* base_ptr_k = (ElementK*)k_traits.base_ptr;
+
+      auto k_traits_cos = static_cast<traits_load_K const&>(params.gmem_tiled_copy_k_cos);
+      ElementK* base_ptr_k_cos = (ElementK*)k_traits_cos.base_ptr;
+
+      auto k_traits_sin = static_cast<traits_load_K const&>(params.gmem_tiled_copy_k_sin);
+      ElementK* base_ptr_k_sin = (ElementK*)k_traits_sin.base_ptr;
+
+      // auto layout_k = gK.layout();
+      constexpr auto static_shape_k = make_shape(size<0>(gK), size<1>(gK));
+      constexpr auto layout_k = make_layout(static_shape_k, LayoutRight{});
+      
+      for (int i =0 ;i< size<2>(gQ) && thread_idx< size<0>(gQ); i++){
+        auto tensorQ = make_tensor(make_gmem_ptr(base_ptr_q+offset_q), layout_q);
+        auto tensorCosQ = make_tensor(make_gmem_ptr(base_ptr_q_cos+offset_q), layout_q);
+        auto tensorSinQ = make_tensor(make_gmem_ptr(base_ptr_q_sin+offset_q), layout_q);
+        cutlass::flash_attention::collective::apply_rope_interleaved_gmem(thread_idx, tensorQ, tensorCosQ, tensorSinQ, tensorQ);
+        offset_q += QK_BLK_M*QK_BLK_K;
+      }
+      
+      for (int i =0 ;i< size<2>(gK) && thread_idx< size<0>(gK); i++){
+        auto tensorK = make_tensor(make_gmem_ptr(base_ptr_k+offset_k), layout_k);
```

Comment:
Looks like `tensorK` will be same for all work-items.

So, if there are multiple workgroups in the grid, there will be a race-condition amongst `i`th work-item of each workgroup while computing `cutlass::flash_attention::collective::apply_rope_interleaved_gmem`.